fix: delete folders using postorder

RafaelPaulovic opened the pull request:
**What is this feature?**  

This PR fixes issue in folder deletion of unified storage implementation to ensure proper cleanup order.

It introduces a new `SortPostorder` utility method that returns folder descendants in postorder (children before parents), ensuring that child folders and their contents are deleted before parent folders.

**Why do we need this feature?**  

When deleting folders with subfolders in the unified storage backend, the deletion order was not guaranteed. This could lead to data drift in legacy and unified due to errors when attempting to delete parent folders before their children are fully cleaned up.

**Who is this feature for?**  

This fix is for all Grafana users who use nested folders and the unified storage backend.

**Which issue(s) does this PR fix?**:  

Fixes: https://github.com/grafana/search-and-storage-team/issues/569

**Special notes for your reviewer:**  

Key changes:
- Added `SortPostorder ` function
- Updated `deleteFromApiServer` and `nestedFolderDelete` to use `SortPostorder ` to ensure correct deletion order (children before parents).
- Maintained backward compatibility by keeping the original `GetDescendants` method unchanged

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

RafaelPaulovic committed:
fix: delete folders using postorder

mustafasencer commented on pkg/services/folder/folderimpl/folder.go:
```diff
@@ -1245,7 +1245,7 @@ func (s *Service) nestedFolderDelete(ctx context.Context, cmd *folder.DeleteFold
 		descendantUIDs = append(descendantUIDs, f.UID)
 	}
-	s.log.InfoContext(ctx, "deleting folder descendants", "org_id", cmd.OrgID, "uid", cmd.UID)
+	s.log.InfoContext(ctx, "deleting folder descendants", "org_id", cmd.OrgID, "uid", cmd.UID, "descendantsUIDs", strings.Join(descendantUIDs, ","))
```
```suggestion
	s.log.InfoContext(ctx, "deleting legacy folder descendants", "org_id", cmd.OrgID, "uid", cmd.UID, "descendantsUIDs", strings.Join(descendantUIDs, ","))
```

mustafasencer commented on pkg/services/folder/folderimpl/folder_unifiedstorage.go:
```diff
@@ -625,6 +625,7 @@ func (s *Service) deleteFromApiServer(ctx context.Context, cmd *folder.DeleteFol
 	}
 	// must delete children first, then the parent folder
 	folders = append(folders, cmd.UID)
+	s.log.InfoContext(ctx, "deleting folders", "org_id", cmd.OrgID, "uid", cmd.UID, "folderUIDs", strings.Join(folders, ","))
```
```suggestion
	s.log.InfoContext(ctx, "deleting folder with descendants", "org_id", cmd.OrgID, "uid", cmd.UID, "descendantsUIDs", strings.Join(folders, ","))
    // must delete children first, then the parent folder
	folders = append(folders, cmd.UID)
```

mustafasencer reviewed (approved):
Thank you! LGTM with minor comments.

RafaelPaulovic committed:
chore: use helper function and do not add method to Folder store

- addresses other review comments fixing log messages and cleans up the unit tests

RafaelPaulovic committed:
chore: run library element tests on modes 2,3,5 only

RafaelPaulovic committed:
chore: adjust to folder.SortByPostorder(folders []*Folder)

RafaelPaulovic committed:
Merge remote-tracking branch 'origin/main' into delete-folders-in-post-order

RafaelPaulovic committed:
chore: run library panels tests in mode 2,3,5 only

RafaelPaulovic committed:
chore: run tests in all modes and increase timeout

- adjusting the modes and tweaking configs will be done separately

mustafasencer reviewed (approved):
LGTM

RafaelPaulovic commented on pkg/services/folder/store.go:
```diff
@@ -48,6 +48,8 @@ type Store interface {
 	// GetDescendants returns all descendants of a folder
 	GetDescendants(ctx context.Context, orgID int64, anchestor_uid string) ([]*Folder, error)
+	// GetDescendantsPostorder returns all descendants of a folder in postorder (children before parents)
+	GetDescendantsPostorder(ctx context.Context, orgID int64, ancestorUID string) ([]*Folder, error)
```
changed to a util function
